configure.ac

```diff
@@ -200,7 +200,7 @@ AC_PATH_PROG(PERL, perl)
 #   -Wall                        CC, CXX
 #
 
-PROSPECTIVE_CFLAGS="-Wall"
+PROSPECTIVE_CFLAGS="-Wall -Werror"
```

Comment:
While we're at it, what about the following flags:
- `-Wextra`
- `-Wshadow`

We should also apply the same flags to `PROSPECTIVE_CXXFLAGS`.

Thoughts?